pkg/plugin/manifest/convert.go

```diff
@@ -129,6 +133,52 @@ func (i Extension) extension(pluginID id.PluginID, sys bool) (*plugin.Extension,
 	return ext, schema, nil
 }
 
+func (l *WidgetLayout) layout() *scene.WidgetLayout {
+	if l == nil {
+		return &scene.WidgetLayout{
+			Extended:   false,
+			Extendable: true,
+			Floating:   false,
+			DefaultLocation: &scene.WidgetLocation{
+				Zone:    "outer",
+				Section: "left",
+				Area:    "top",
+			},
+		}
+	}
+
+	var swl scene.WidgetLayout
+	swl.Extended = false
+	if l.Extendable == nil {
+		swl.Extendable = true
+	} else {
+		e := l.Extendable
+		swl.Extendable = *e
+	}
+	if l.Floating == nil {
+		swl.Floating = false
+	} else {
+		f := l.Floating
+		swl.Floating = *f
+	}
+	var dl scene.WidgetLocation
```

Comment:
```suggestion
	var dl *scene.WidgetLocation
```

Be careful of diff between reference and vale. This code always generates empty WidgetLocation struct (not nil) and it will be set to scene.